src/tests/e2e/auth/login.test.ts

```diff
@@ -0,0 +1,124 @@
+
+	test.describe("Client-Side Validation", () => {
+		test("should not be able to log-in with empty email", async ({ page }) => {
+			await page.goto(`${baseUrl}/map`);
+			await page.getByRole("link", { name: "Profil" }).click();
+			await page.getByLabel("Deine E-Mail Adresse").click();
+			await page.getByLabel("Deine E-Mail Adresse").fill(" ");
+			await page.getByLabel("Deine E-Mail Adresse").press("Backspace");
+			await page.getByLabel("Deine E-Mail Adresse").press("Tab");
+			await page.getByLabel("Passwort").fill("123");
+			await page.getByLabel("Passwort").press("Enter");
+
+			await expect(page.locator("input#email:invalid")).toBeVisible();
+		});
+
+		test("should not be able to log-in with invalid email format", async ({
+			page,
+		}) => {
+			await page.goto(`${baseUrl}/map`);
+			await page.getByRole("link", { name: "Profil" }).click();
+			await page.getByLabel("Deine E-Mail Adresse").click();
+			await page.getByLabel("Deine E-Mail Adresse").fill("invalid-email");
+			await page.getByLabel("Deine E-Mail Adresse").press("Tab");
+			await page.getByLabel("Passwort").fill("123");
+			await page.getByLabel("Passwort").press("Enter");
+
+			await expect(page.locator("input#email:invalid")).toBeVisible();
+		});
+
+		test("should not be able to log-in with empty password", async ({
+			page,
+		}) => {
+			await page.goto(`${baseUrl}/map`);
+			await page.getByRole("link", { name: "Profil" }).click();
+			await page.getByLabel("Deine E-Mail Adresse").click();
+			await page.getByLabel("Deine E-Mail Adresse").fill("invalid-email");
+			await page.getByLabel("Deine E-Mail Adresse").press("Tab");
+			await page.getByLabel("Passwort").fill(" ");
+			await page.getByLabel("Passwort").press("Backspace");
+			await page.getByLabel("Passwort").press("Enter");
+
+			await expect(page.locator("input#password:invalid")).toBeVisible();
+		});
+	});
+
+	test.describe("Server-Side Validation", () => {
+		test("should not be able to log-in with wrong email/password credentials", async ({
+			page,
+		}) => {
+			await page.goto(`${baseUrl}/map`);
+			await page.getByRole("link", { name: "Profil" }).click();
+			await page.getByLabel("Deine E-Mail Adresse").click();
+			await page
+				.getByLabel("Deine E-Mail Adresse")
+				.fill("email-with-no-account@example.com");
+			await page.getByLabel("Deine E-Mail Adresse").press("Tab");
+			await page.getByLabel("Passwort").fill("invalid-password");
+			await page.getByLabel("Passwort").press("Enter");
+			await expect(
+				page
+					.locator("div")
+					.filter({ hasText: /^Falsches Passwort oder E-Mail Adresse$/ })
+					.nth(2),
+			).toBeVisible();
+		});
+	});
+
+	test.describe("Error Handling", () => {
+		test("should show error toast when and unexpected error occurs", async ({
+			browser,
+		}) => {
+			const browserContext = await browser.newContext();
+			const page = await browserContext.newPage();
+
+			await page.goto(`${baseUrl}/profile`);
+
+			await page.getByLabel("E-mail").click();
```

Comment:
not test related, but I see mixed versions of "email" spelling:

- `E-mail` and sometimes `E-Mail`

Reply:
good catch 👍  I just realized the e-mail field is not using a standard i18n wording and has hardcoded values here and there. will fix 👍

Reply:
fixed it 👍